src/logic/wallets/getWeb3.ts

```diff
@@ -64,39 +66,51 @@ export const getInfuraUrl = () => {
 export const web3ReadOnly =
   process.env.NODE_ENV !== 'test'
     ? new Web3(new Web3.providers.HttpProvider(getInfuraUrl()))
-    : new Web3((window as any).web3.currentProvider)
+    : new Web3(window.web3?.currentProvider || 'ws://localhost:8545')
 
 let web3 = web3ReadOnly
-export const getWeb3 = () => web3
+export const getWeb3 = (): Web3 => web3
 
-export const resetWeb3 = () => {
+export const resetWeb3 = (): void => {
   web3 = web3ReadOnly
 }
 
-export const getAccountFrom = async (web3Provider) => {
+export const getAccountFrom = async (web3Provider: Web3): Promise<string | null> => {
   const accounts = await web3Provider.eth.getAccounts()
 
-  if (process.env.NODE_ENV === 'test' && (window as any).testAccountIndex) {
-    return accounts[(window as any).testAccountIndex]
+  if (process.env.NODE_ENV === 'test' && window.testAccountIndex) {
+    return accounts[window.testAccountIndex]
   }
 
   return accounts && accounts.length > 0 ? accounts[0] : null
 }
 
-export const getNetworkIdFrom = (web3Provider) => web3Provider.eth.net.getId()
+export const getNetworkIdFrom = (web3Provider: Web3): Promise<number> => web3Provider.eth.net.getId()
 
-const isHardwareWallet = (walletName) =>
+const isHardwareWallet = (walletName: string) =>
   sameAddress(WALLET_PROVIDER.LEDGER, walletName) || sameAddress(WALLET_PROVIDER.TREZOR, walletName)
 
-const isSmartContractWallet = async (web3Provider, account) => {
+const isSmartContractWallet = async (web3Provider: Web3, account) => {
```

Comment:
let's also type `account` and add a return type :)

Reply:
done!